cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -185,12 +185,14 @@ static void gpi_log_v(const char *name, enum gpi_log_levels level, const char *p
         goto error;
     }
 
-    PyObject *filter_ret = PyObject_CallFunctionObjArgs(pLogFilter, logger_name_arg, level_arg, NULL);
+    PyObject *filter_ret;
+    filter_ret = PyObject_CallFunctionObjArgs(pLogFilter, logger_name_arg, level_arg, NULL);
```

Comment:
This might look a little weird, but it makes sense. You cannot jump over a variable initialization, but you can jump over variable declarations *if* they have trivial default constructors. Value initialization calls a non-default constructor, so it falls into the "ill-formed" category. The language could probably support what we are trying to do by checking for a trivial default constructor and trivial assignment, but it doesn't...